elements/lisk-p2p/src/utils/misc.ts

```diff
@@ -126,10 +124,10 @@ export const getBucket = (options: {
 	readonly secret: number;
 	readonly peerType: PEER_TYPE;
 	readonly targetAddress: string;
+	readonly bucketCount: number;
 }): number => {
-	const { secret, targetAddress, peerType } = options;
+	const { secret, targetAddress, peerType, bucketCount } = options;
```

Comment:
I'm not sure if it works fine with any value of bucket count because so far we are using `64` and `128`. Maybe we can cover it up in unit tests?

Reply:
I already updated the unit tests to test this and removed the stubs.

Reply:
I meant, tests with different values of `bucketCount` and see if the bucket number is returned within the range. I only see tests covering for `MAX_TRIED_BUCKETS` which is `128` Let's say having a test if choose a `bucketCount` of 12 and see if `getBucket` returns a number between 0 and 11

Reply:
I think it will work. It seems @jondubois added `bucketCount` for unit test purposes only, and in the actual code it will always be `64` and `124`. The edge case in which it will not work is if `0` is passed.

Reply:
If we create a new peer list and use a value other than 64 or 128 and if the function return a value outside the given `bucketCount` then it could be a problem. If we are giving `bucketCount` required value to this function then we have to make sure it should not only work for limited values. It will be nice if we can have a unit test covering a few random values of `bucketCount`.

Reply:
@ishantiw It's a good point I don't see the need to change `secondMod` to `bucketCount` as the tests only used `MAX_TRIED_BUCKETS` and in the code it will always be `64` or `128`. This is why I originally had it as using `secondMod`.  However I don't think it's necessary to add more unit tests to check if other values would work.

Reply:
I made the secondMod configurable because the higher level PeerBucket logic assumed that the getBucket function allowed this. It could be useful in the future if we want to make the bucket values configurable.

Right now the distribution of the `getBucket` function is tested with two different numbers: 64 and 128; if it works with both of those then it will work with any other value because the second modulo expression doesn't treat those two numbers as special. I agree with @mitsujutsu that the main special case is 0.

Reply:
A reminder to add the check for `0` `bucketCount` value as discussed above